src/controller/python/matter/clusters/Attribute.py

```diff
@@ -545,11 +547,27 @@ def Callback(transaction: SubscriptionTransaction)
     def SetAttributeUpdateCallback(self, callback: Callable[[TypedAttributePath, SubscriptionTransaction], None]):
         '''
         Sets the callback function for the attribute value change event,
-        accepts a Callable accepts an attribute path and the cached data.
+        accepts a Callable that accepts an attribute path and the cached data.
         '''
         if callback is not None:
             self._onAttributeChangeCb = callback
 
+    def SetReportBeginCallback(self, callback: Callable[[SubscriptionTransaction], None]):
+        '''
+        Sets the callback function for when a subscription report with at least one path starts,
+        accepts a Callable that accepts the transaction.
+        '''
+        if callback is not None:
```

Comment:
NIT: Filtering None will silently ignore programming errors. IMHO, it's better to trow exception here (and in other setters) if the type is incorrect.

Reply:
Done